### ISSUE TITLE
Using exported types for TypeScript

### ISSUE THREAD

**Issue by TomlDev** [bkrem/react-d3-tree#231]
## Are you reporting a bug, or opening a feature request?
feature request
  
## What is the actual behavior/output?
I cannot use the types (eg. NodeSvgShape/ReactD3TreeItem) because they don't get exported.

## What is the behavior/output you expect?
That I can use the react-d3-tree types in my project for type-validation

## What version of react-d3-tree are you using?
latest

**Comment by bkrem** [comment]
Hi @TomlDev, thanks for pointing that out, oversight on my part in terms of exporting all relevant types when I set up the definition file.

It should be fixed now in `v1.14.1` (https://github.com/bkrem/react-d3-tree/commit/61e3351ce61dbfb421d5de14ad17a1085a0e173c) 

Feel free to reopen if there's still an issue.

**Comment by TomlDev** [comment]
Thanks @bkrem !
Another thing that occurs (and that was the initial reason I needed those types) is that onUpdate option passes a ReactD3TreeItem as first parameter, but that data also includes zoom & translate attributes which don't occur in ReactD3TreeItem.. Shall I open a new issue for that?

**Comment by bkrem** [comment]
Oh wow yeah great point, that's been wrong all along 🤦‍♂ this is the issue with retrofitting typings onto an old codebase.

Thanks for pointing that out too 💯  I've updated the type declarations and the docs (https://github.com/bkrem/react-d3-tree/commit/ac7ba4fbd48799653c2a05be9cae2bdcb41ddb83) and published as `v1.14.2`.

I had a quick play around with it, but feel free to open another issue if the types are still problematic in some way.

**Comment by TomlDev** [comment]
Thanks @bkrem, I get this error after updating:
```
TypeError: a is null
value
node_modules/react-d3-tree/lib/react-d3-tree.min.js:1024

  1021 | 
  1022 | if (n) {
  1023 |   var i = "object" === (void 0 === a ? "undefined" : u(a)),
> 1024 |       s = i ? a.x : 0,
       | ^  1025 |       l = i ? a.y : 0;
  1026 |   return "horizontal" === t ? "translate(" + l + "," + s + ")" : "translate(" + s + "," + l + ")";
  1027 | }

t
node_modules/react-d3-tree/lib/react-d3-tree.min.js:970

  967 | }
  968 | 
  969 | return n = r = a(this, (e = t.__proto__ || Object.getPrototypeOf(t)).call.apply(e, [this].concat(l))), r.state = {
> 970 |   transform: r.setTransform(r.props.nodeData, r.props.orientation, !0),
      | ^  971 |   initialStyle: {
  972 |     opacity: 0
  973 |   }
```
